nipoppy_cli/docs/source/user_guide/processing.md

```diff
@@ -0,0 +1,132 @@
+# Running processing pipelines
+
+Just like with the BIDS conversion pipelines, Nipoppy uses the {term}`Boutiques framework <Boutiques>` to run image processing pipelines. By default, new Nipoppy datasets (as created with [`nipoppy init`](<project:../cli_reference/init.md>)) are populated with descriptor files and default invocation files for the following processing pipelines:
+- [fMRIPrep](https://fmriprep.org/en/stable/), a pipeline for preprocessing anatomical and functional MRI data.
+- [MRIQC](https://mriqc.readthedocs.io/en/latest/index.html), a pipeline for automated quality control (QC) metric extraction
+
+```{note}
+Although fMRIPrep and MRIQC are both [BIDS Apps](https://bids-apps.neuroimaging.io/about/), Nipoppy can also be used to run pipelines that are not BIDS Apps. Custom pipelines can be added by creating a Boutiques descriptor file and modifying the global configuration file accordingly.
+```
+
+## Summary
+
+### Prerequisites
+
+- A Nipoppy dataset with a valid global configuration file and an accurate manifest
+    - See the [Quickstart guide](../quickstart.md) for instructions on how to set up a new dataset
+- Raw imaging data organized according to the {term}`BIDS` standard in the {{dpath_bids}} directory
+    - See <project:bids_conversion.md>
+
+```{include} ./inserts/apptainer_stub.md
+```
+
+### Data directories
+
+| Directory | Content description |
+|---|---|
+| {{dpath_bids}} | **Input** -- {{content_dpath_bids}} |
+| {{dpath_derivatives}} | **Output** -- {{content_dpath_derivatives}} |
+
+Within the {{dpath_derivatives}} directory, output files for a specific pipeline are organized like this:
+```{literalinclude} ./inserts/pipeline_derivatives.txt
+---
+class: no-copybutton
+---
+```
+
+### Commands
+
+- Command-line interface: [`nipoppy run`](<project:../cli_reference/run.md>)
+- Python API: {class}`nipoppy.workflows.PipelineRunner`
+
+### Workflow
+
+1. Nipoppy will loop over all participants/sessions that *have* BIDS data according to the {term}`doughnut file` but *have not* yet successfully completed the pipeline according the the imaging derivatives bagel file
+    - An existing, out-of-date doughnut file can be updated with [`nipoppy doughnut --regenerate`](../cli_reference/doughnut.md)
+2. For each participant-session pair:
+    1. The pipeline's invocation will be processed such that template strings related to the participant/session and dataset paths are replaced by the appropriate values
+    2. A [PyBIDS](https://bids-standard.github.io/pybids/) database indexing the BIDS data for this participant and session is created in a subdirectory inside {{dpath_bids_db}}
```

Comment:
Is this only for BIDS apps or regardless?

Reply:
Right now: regardless. But I think I created an issue to make this optional (even for BIDS apps)